tests/parallelcluster-release-check.py

```diff
@@ -90,138 +128,396 @@ def _double_writeln(fileo, message):
     fileo.write(message + "\n")
 
 
-# Helper method to get the name of the autoscaling group
-def check_asg_capacity(stack_name, region, out_f):
-    asg_conn = boto3.client("autoscaling", region_name=region)
-    iter = 0
-    capacity = -1
-    while iter < 24 and capacity != 0:
-        try:
-            r = asg_conn.describe_tags(Filters=[{"Name": "value", "Values": [stack_name]}])
-            asg_name = r.get("Tags")[0].get("ResourceId")
-            response = asg_conn.describe_auto_scaling_groups(AutoScalingGroupNames=[asg_name])
-            capacity = response["AutoScalingGroups"][0]["DesiredCapacity"]
-            iter += 1
-            time.sleep(10)
-        except Exception as e:
-            _double_writeln(out_f, "check_asg_capacity failed with %s exception: %s" % (type(e), e))
-            raise
+def _get_attached_compute_nodes(cluster_config):
+    """
+    Returns the number of compute nodes attached to the scheduler.
+    Args:
+        cluster_config: named tuple of type ClusterConfig containing the configuration of the cluster.
+
+    Returns:
+        number_of_nodes: number of available compute nodes.
+    """
+    output = _exec_ssh_command(
+        command=get_compute_nodes_command_map[cluster_config.scheduler],
+        username=cluster_config.username,
+        host=cluster_config.master_node,
+        key_path=cluster_config.key_path,
+    )
+    # get last line of the output containing the number of compute nodes
+    return int(output.split()[-1])
```

Comment:
We could detect `_exec_ssh_command` failures or catch the `TypeError` in the `int()` conversion.

Reply:
We agreed that the global try-catch in the run-test function is enough to record such errors.